python/tests/operations/slice.py

```diff
@@ -0,0 +1,768 @@
+        self.assertTrue(
+            isinstance(
+                slice.slice(tensorblock, samples_to_slice=samples_to_slice), TensorBlock
+            )
+        )
+        # Check 3: passing tensor=float raises error
+        with self.assertRaises(ValueError):
+            slice.slice(5.0, samples_to_slice=samples_to_slice)
+        # Check 4: passing samples_to_slice=np.array raises error
+        with self.assertRaises(AssertionError):
+            slice.slice(
+                tensormap,
+                samples_to_slice=np.array(
+                    [
+                        [
+                            5,
+                        ],
+                        [
+                            6,
+                        ],
+                    ]
+                ),
+            )
+
+    # TEST BLOCK 7: TESTING WARNINGS OF THE SLICE FUNCTION
+
+    def test_slice_samples_tensormap_partially_empty_warning(self):
+        tensormap = equistore.io.load(
+            os.path.join(DATA_ROOT, TEST_FILE),
+            use_numpy=True,
+        )
+        # Remove the gradients to simplify test
+        tensormap = remove_gradients(tensormap)
+        # Slice only 'structures' 2
+        structures_to_keep = np.array([2]).reshape(-1, 1)
+        samples_to_slice = Labels(
+            names=["structure"],
+            values=structures_to_keep,
+        )
+        with self.assertWarns(Warning):
+            # Check 1: warning raised as some empty blocks produced
+            slice.slice(
+                tensormap,
+                samples_to_slice=samples_to_slice,
+            )
+
+    def test_slice_samples_tensormap_completely_empty_warning(self):
+        tensormap = equistore.io.load(
+            os.path.join(DATA_ROOT, TEST_FILE),
+            use_numpy=True,
+        )
+        # Remove the gradients to simplify test
+        tensormap = remove_gradients(tensormap)
+        # Slice only 'structures' -1 (i.e. a sample that doesn't exist in the data)
+        structures_to_keep = np.array([-1]).reshape(-1, 1)
+        samples_to_slice = Labels(
+            names=["structure"],
+            values=structures_to_keep,
+        )
+        with self.assertWarns(Warning):
+            # Check 1: warning raised as all empty blocks produced
+            slice.slice(
+                tensormap,
+                samples_to_slice=samples_to_slice,
+            )
+
+    def test_slice_properties_tensormap_completely_empty_warning(self):
+        tensormap = equistore.io.load(
+            os.path.join(DATA_ROOT, TEST_FILE),
+            use_numpy=True,
+        )
+        # Remove the gradients to simplify test
+        tensormap = remove_gradients(tensormap)
+        # Slice only 'n' (i.e. radial channels) -1 (i.e. non-existent channel)
+        channels_to_keep = np.array([-1]).reshape(-1, 1)
+        properties_to_slice = Labels(
+            names=["n"],
+            values=channels_to_keep,
+        )
+        with self.assertWarns(Warning):
```

Comment:
you can also check the warnings message. Check for tests with assertRaises for similar example

Reply:
Done in commit 434abeaa6a8d1b532e8ee011926cb630b336c4da.